### PR TITLE
Call unlockArticle if provided when consuming showcaseEntitlement

### PR DESCRIPTION
**Pull request by teiswit**
Previously (#2771) we made `unlockArticle` optional but not really calling it in a server-side paywall scenario (i.e. when showcaseEntitlement is provided). This change will make `unlockArticle` effective when provided.

- Publishers who unlock articles on the server-side will not be impacted by this change. GAA will initialize a default `unlockArticle: () => {}`.
- Publishers who provide `unlockArticle` will have this method called when `showcaseEntitlement` is consumed.
 
Ref: b/234314583#5

### CODE REVIEW

File: src/runtime/extended-access/gaa-metering.js [hunk 1]
```diff
@@ -145,7 +145,11 @@ export class GaaMetering {
         unlockArticleIfGranted();
       } else if (showcaseEntitlement) {
         debugLog(showcaseEntitlement);
-        subscriptions.consumeShowcaseEntitlementJwt(showcaseEntitlement);
+        subscriptions.consumeShowcaseEntitlementJwt(showcaseEntitlement, () => {
```
Comment (ChrisAntaki) [inline src/runtime/extended-access/gaa-metering.js:148]: Could you move the definition of the `unlockArticle` variable here? The new code below looks like the first usage of the variable

**Review by ChrisAntaki — CHANGES_REQUESTED** [review]
Looking good, I just suggested moving a variable definition to improve clarity

File: src/runtime/extended-access/gaa-metering.js (hunk 1 above)
Comment (teiswit) [reply src/runtime/extended-access/gaa-metering.js:148]: It's actually being used prior to this line in a `unlockArticleIfGranted`, so I've moved the definition of `unlockArticle` closest to where it's first used.

**Comment by teiswit** [comment]
Please take another look. Thanks!

File: src/runtime/extended-access/gaa-metering.js (hunk 1 above)
Comment (ChrisAntaki) [reply src/runtime/extended-access/gaa-metering.js:148]: Thanks! And sorry for missing that

**Review by ChrisAntaki — APPROVED** [review]

**Pull request merged**